File: src/main/java/org/stevenw/shops/ShopItem.java
package org.stevenw.shops;

import org.bukkit.inventory.ItemStack;
import org.stevenw.shops.economy.VulcanEconomyConnector;

public class ShopItem {
    private ItemStack item;
    private Double price;
    private boolean legacy;
    private int amount;
    private String currencyKey;
    private String currencyName;

    public ShopItem(ItemStack item, Double price, boolean useLegacy) {
        this(item, price, useLegacy, 1);
    }
    public ShopItem(ItemStack item, Double price, boolean useLegacy, int amount) {
        this(item, price, useLegacy, amount, null);
    }
    public ShopItem(ItemStack item, Double price, boolean useLegacy, int amount, String currencyKey) {
        this.item = item;
        this.price = price;
        this.legacy = useLegacy;
        this.amount = amount;
        if(currencyKey == null) {
            if(sShops.plugin.getEconomy() instanceof VulcanEconomyConnector) {
                this.currencyKey = ((VulcanEconomyConnector) sShops.plugin.getEconomy()).getDefaultCurrencyKey();
                this.currencyName = ((VulcanEconomyConnector) sShops.plugin.getEconomy()).getDefaultCurrencyName();
            } else {
                this.currencyKey = null;
                this.currencyName = "Dollars";
            }
        } else {
            this.currencyKey = currencyKey;
            if(sShops.plugin.getEconomy() instanceof VulcanEconomyConnector) {
                this.currencyName = ((VulcanEconomyConnector) sShops.plugin.getEconomy()).getCurrencyName(currencyKey);
            } else {
                this.currencyName = "Dollars";
            }
        }
    }
    public ShopItem(ItemStack item, Double price) {
        this(item, price, false);
    }
    public ItemStack getItem() {
        return this.item;
    }
    public Double getPrice() {
        return this.price;
    }

    public boolean isLegacy() {
        return legacy;
    }

    public int getAmount() {
        return amount;
    }

    public String getCurrencyKey() {
        return currencyKey;
    }

    public String getCurrencyName() {
        return currencyName;
    }
}

File: src/main/java/org/stevenw/shops/items/ShopItem.kt
package org.stevenw.shops.items

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.stevenw.shops.economy.VulcanEconomyConnector
import org.stevenw.shops.sShops
import org.stevenw.shops.types.BuyShop

abstract class ShopItem(val config : ConfigurationSection, val item: ItemStack, val price: Double?, val
amount: Int, currencyKey: String?) {
    var currencyKey: String? = null
        private set
    var currencyName: String? = null
        private set

    init {
        addCurrencies()
        //this.lore = "Price: " + price!!.toString() + " " + currencyName + if (amount == 1) "" else " " + " for $"

    }


    abstract fun execute(buyShop:BuyShop, player: Player);
    abstract fun getLore() : String;


    private fun addCurrencies() {
        if (currencyKey == null) {
            if (sShops.plugin.economy is VulcanEconomyConnector) {
                this.currencyKey = (sShops.plugin.economy as VulcanEconomyConnector).defaultCurrencyKey
                this.currencyName = (sShops.plugin.economy as VulcanEconomyConnector).defaultCurrencyName
            } else {
                this.currencyKey = null
                this.currencyName = "Dollars"
            }
        } else {
            this.currencyKey = currencyKey
            if (sShops.plugin.economy is VulcanEconomyConnector) {
                this.currencyName = (sShops.plugin.economy as VulcanEconomyConnector).getCurrencyName(currencyKey)
            } else {
                this.currencyName = "Dollars"
            }
        }
    }
}
